src/interactive/azext_interactive/azclishell/scenario_suggest.py

```diff
@@ -78,15 +118,25 @@ def try_remove(container, item):
                         if self.param_value_map[param]:
                             suggest.append(self.param_value_map[param])
                     return Suggestion(' '.join(suggest))
-            # If the user finish input with space, suggest all the rest unused parameters
             elif unfinished == '':
                 if not last_part.startswith('-'):
-                    suggest = []
+                    suggests = []
                     for param in unused_param:
-                        suggest.append(param)
-                        if self.param_value_map[param]:
-                            suggest.append(self.param_value_map[param])
-                    return Suggestion(' '.join(suggest))
+                        if param in self.special_global_param_map.keys():
+                            # sample_value is the sample values in scenarios, such as <RESOURCEGROUPNAME>
+                            sample_value = param
+                        else:
+                            sample_value = self.param_value_map[param]
+                        if sample_value:
+                            if sample_value in self.customized_param_value_map.keys():
+                                value = self.customized_param_value_map[sample_value]
+                            else:
+                                value = ''
+                            suggests.append({'param': param, 'value': value})
+                    if suggests:
+                        # suggest one parameter at a time
+                        return Suggestion(' '.join([suggests[0]['param'], suggests[0]['value']]))
```

Comment:
It is because suggests contains a list of recommendations ranging from comands to scenarios, do you mean to use the option_index such as “a1” to the key of dictionary `suggests` and suggested data to be value of the dict?

Reply:
We no longer need to define `suggests` as a list, right?